src/inmanta/util/dict_path.py

```diff
@@ -710,7 +710,8 @@ class WildComposedPath(WildDictPath):
     """
 
     element_types: Sequence[type[WildDictPath]] = [WildInDict, WildKeyedList]
-    COMPOSED_DICT_PATH_PATTERN = re.compile(r"(?:[^.\\]|\\.)+")
+    KEYED_LIST_FILTER = r"(?:[^.\\]+(?<!\\)(?:\[.*?(?<!\\)\])+)"
```

Comment:
1. Isn't the first `(?<!\\)` redundant? From the docs I get that it's a negative lookbehind, but the previous character is `[^.\\]`.
2. ~What's the question mark at `.*?`?~ non-greedy matching, good.
3. I think we can safely simplify some more. The regex below only seems to allow escapes for dots anyway, so let's stick to trying to match `[...]` here, not `name[...]`, and don't try to account for escaped `]`. And don't allow backslashes except for dots, as below. And then we can unify some more: use this one in the same repeating "or" that is already used in the regex below.